File: backend/app/utils/text_normalization.py
"""Text normalization utilities for form inputs."""


def normalize_form_text(text: str | None) -> str | None:
    """
    Normalize text input for forms:
    - First letter uppercase
    - Rest lowercase

    Example: "MARKET ALIŞVERIŞI" → "Market alışverişı"
    """
    if not text or not isinstance(text, str):
        return text

    trimmed = text.strip()
    if not trimmed:
        return ""

    return trimmed[0].upper() + trimmed[1:].lower()


def normalize_form_data(data: dict, fields_to_normalize: list[str]) -> dict:
    """
    Apply normalization to multiple fields in a dictionary.
    """
    normalized = data.copy()
    for field in fields_to_normalize:
        if field in normalized and isinstance(normalized[field], str):
            normalized[field] = normalize_form_text(normalized[field])
    return normalized

File: frontend/src/utils/textNormalization.ts
/**
 * Normalize text input for forms:
 * - First letter/word uppercase
 * - Rest lowercase
 * Examples: "MARKET ALIŞVERIŞI" → "Market alışverişı"
 */
export function normalizeFormText(text: string): string {
  if (!text || typeof text !== 'string') return ''
  const trimmed = text.trim()
  if (trimmed.length === 0) return ''
  return trimmed.charAt(0).toUpperCase() + trimmed.slice(1).toLowerCase()
}

/**
 * Apply normalization to multiple fields
 */
export function normalizeFormData(data: Record<string, any>, fieldsToNormalize: string[]): Record<string, any> {
  const normalized = { ...data }
  fieldsToNormalize.forEach(field => {
    if (field in normalized && typeof normalized[field] === 'string') {
      normalized[field] = normalizeFormText(normalized[field])
    }
  })
  return normalized
}
